scripts/brc4/gff-overlaps.py

```diff
@@ -24,138 +24,166 @@
-#        print( type(rec) )
-#        print(rec.id)
-        seqname = str( rec.id )
+
+    for rec in GFF.parse(in_handle, limit_info=limit_info):
+        #        print( type(rec) )
+        #        print(rec.id)
+        seqname = str(rec.id)
         if seqname not in seqDict:
-            seqDict[ seqname ]['plus'] = []
-            seqDict[ seqname ]['minus'] = []
-            
-        get_intervals( rec, genesDict, seqDict, seqname )    
-                 
+            seqDict[seqname]["plus"] = []
+            seqDict[seqname]["minus"] = []
+
+        get_intervals(rec, genesDict, seqDict, seqname)
+
     in_handle.close()
 
-    overlaps = write_report( out_file, seqDict, genesDict )
-    
+    overlaps = write_report(out_file, seqDict, genesDict)
+
     msg_total_genes = f"total genes scanned = { len(genesDict) }"
-    print( msg_total_genes )
-    logging.info( msg_total_genes )
-    
+    print(msg_total_genes)
+    logging.info(msg_total_genes)
+
     msg_total_overlaps = f"total overlaps detected = { overlaps }"
-    print( msg_total_overlaps )
-    logging.info( msg_total_overlaps ) 
+    print(msg_total_overlaps)
+    logging.info(msg_total_overlaps)
 
-    logging.info( "stop" )
+    logging.info("stop")
     sys.exit(0)
-            
-if __name__ == '__main__':
-
-    parser = argparse.ArgumentParser(description='Scan a GFF file to detect overlapping seqfeature objects (default = gene).')
-    parser.add_argument('--gff', help='path of gff file to scan for overlaps', required=True, nargs='?' )
-    parser.add_argument('--stats', help='run stats analysis of GFF file instead of parsing GFF file', action='store_true' )
-    parser.add_argument('--logfile', help='path of file for logging (default=log.overlaps)', nargs='?', default='log.overlaps' )
-    parser.add_argument('--output', help='path of file for output (default=overlaps)', required=True, nargs='?', default='overlaps' )
-    parser.add_argument('--filter', help='seqfeature type to use for overlap isolation (default=gene)', nargs='?', default='gene' )
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(
+        description="Scan a GFF file to detect overlapping seqfeature objects (default = gene)."
+    )
+    parser.add_argument("--gff", help="path of gff file to scan for overlaps", required=True, nargs="?")
+    parser.add_argument(
+        "--stats", help="run stats analysis of GFF file instead of parsing GFF file", action="store_true"
+    )
+    parser.add_argument(
+        "--logfile", help="path of file for logging (default=log.overlaps)", nargs="?", default="log.overlaps"
+    )
+    parser.add_argument(
+        "--output",
+        help="path of file for output (default=overlaps)",
+        required=True,
+        nargs="?",
+        default="overlaps",
+    )
+    parser.add_argument(
+        "--filter",
+        help="seqfeature type to use for overlap isolation (default=gene)",
+        nargs="?",
+        default="gene",
+    )
```

Comment:
It's a bit weird but that's what the line limit does... I would actually be ok with them all being multiline, at least that'd be uniform (we can force that by adding a comma after the last element, cf last 2 add_argument)

Reply:
In my opinion it may make some files significantly longer, but I do not mind either way.